### PR TITLE
Env var override

### PR DESCRIPTION
**Pull request by ankitsinghal130290**

### CODE REVIEW

File: src/main/java/com/microsoft/kafkaavailability/PropertiesManager.java [hunk 1]
```diff
@@ -82,22 +83,31 @@ public void setProperty(String propName,String override){
...
-            if(dataType == LIST_TYPE){
-                List<String> value = new ArrayList<String>(Arrays.asList(override.split(",")));
-                set(field,value);
-            }
-            if(dataType == INT_TYPE){
-                int value = Integer.parseInt(override);
-                set(field,value);
+            switch (dataType){
+                case LIST_TYPE:
+                    List<String> value = new ArrayList<String>(Arrays.asList(override.split(",")));
```
Comment (luigiw) [inline src/main/java/com/microsoft/kafkaavailability/PropertiesManager.java]: You don't need the constructor, Arrays.aslist returns Arraylist already.

File: src/main/java/com/microsoft/kafkaavailability/PropertiesManager.java [hunk 2]
```diff
@@ -51,7 +51,7 @@ public PropertiesManager(String propFileName, Class<T> typeParameterClass) throw
         {
             String text = Resources.toString(url, Charsets.UTF_8);
             m_prop = gson.fromJson(text, m_typeParameterClass);
-            MergePropsFromEnv(m_prop);
+            this.MergePropsFromEnv(m_prop);
```
Comment (soumyajit-sahu) [inline src/main/java/com/microsoft/kafkaavailability/PropertiesManager.java]: Can we change this method name to start with lower case?

Comment (luigiw) [inline src/main/java/com/microsoft/kafkaavailability/PropertiesManager.java]: +1   :)

File: src/main/java/com/microsoft/kafkaavailability/PropertiesManager.java [hunk 3]
```diff
@@ -82,22 +83,31 @@ public void setProperty(String propName,String override){
...
+                    set(field,value);
+                    break;
+                case INT_TYPE:
+                    int intData = Integer.parseInt(override);
+                    set(field,intData);
+                    break;
+                case STRING_TYPE:
+                    set(field,override);
+                    break;
+                default:
```
Comment (soumyajit-sahu) [inline src/main/java/com/microsoft/kafkaavailability/PropertiesManager.java]: Would it be worthwhile to add the cases for LONG, DOUBLE and BOOLEAN while we are here?

File: src/main/java/com/microsoft/kafkaavailability/PropertiesManager.java (hunk 2 above)
Comment (ankitsinghal130290) [inline src/main/java/com/microsoft/kafkaavailability/PropertiesManager.java]: I tried to follow the convention which is already in the file as other method starts with a capital letter. But I can change it to lower case if that is a hard convention.

Comment (luigiw) [inline src/main/java/com/microsoft/kafkaavailability/PropertiesManager.java]: It's just a more common convention for Java code to name methods starting with lower case character, and I've been trying to change methods name to start with lower case character when I'm making codes changes.

Comment (ankitsinghal130290) [inline src/main/java/com/microsoft/kafkaavailability/PropertiesManager.java]: This is done I confused it with other codebase.

File: src/main/java/com/microsoft/kafkaavailability/PropertiesManager.java (hunk 1 above)
Comment (ankitsinghal130290) [inline src/main/java/com/microsoft/kafkaavailability/PropertiesManager.java]: Done thanks for the lookout.

File: src/main/java/com/microsoft/kafkaavailability/PropertiesManager.java (hunk 3 above)
Comment (ankitsinghal130290) [inline src/main/java/com/microsoft/kafkaavailability/PropertiesManager.java]: Added this , this should be useful.

**Pull request merged**